src/Numerics/Distributions/BetaBinomial.cs

```diff
@@ -0,0 +1,416 @@
+        /// </summary>
+        /// <param name="n">The number of Bernoulli trials n - n is a positive integer </param>
+        /// <param name="a">Shape parameter alpha of the Beta distribution. Range: a > 0.</param>
+        /// <param name="b">Shape parameter beta of the Beta distribution. Range: b > 0.</param>
+        /// <param name="randomSource">The random number generator which is used to draw random samples.</param>
+        public BetaBinomial(int n, double a, double b, System.Random randomSource)
+        {
+            if (!IsValidParameterSet(n,a,b))
+            {
+                throw new ArgumentException(Resources.InvalidDistributionParameters);
+            }
+
+            _random = randomSource ?? SystemRandomSource.Default;
+            _n = n;
+            _a = a;
+            _b = b;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="System.String"/> that represents this instance.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="System.String"/> that represents this instance.
+        /// </returns>
+        public override string ToString()
+        {
+            return $"BetaBinomial(n = {_n}, a = {_a}, b = {_b})";
+        }
+
+        /// <summary>
+        /// Tests whether the provided values are valid parameters for this distribution.
+        /// </summary>
+        /// <param name="n">The number of Bernoulli trials n - n is a positive integer </param>
+        /// <param name="a">Shape parameter alpha of the Beta distribution. Range: a > 0.</param>
+        /// <param name="b">Shape parameter beta of the Beta distribution. Range: b > 0.</param>
+        public static bool IsValidParameterSet(int n, double a, double b)
+        {
+            return n >= 1.0 && a > 0.0 && b > 0.0;
+        }
+
+        /// <summary>
+        /// Tests whether the provided values are valid parameters for this distribution.
+        /// </summary>
+        /// <param name="n">The number of Bernoulli trials n - n is a positive integer </param>
+        /// <param name="a">Shape parameter alpha of the Beta distribution. Range: a > 0.</param>
+        /// <param name="b">Shape parameter beta of the Beta distribution. Range: b > 0.</param>
+        /// <param name="k">The location in the domain where we want to evaluate the probability mass function.</param>
+        public static bool IsValidParameterSet(int n, double a, double b, int k)
+        {
+            return n >= 1.0 && a > 0.0 && b > 0.0 && k >=0 && k <=n;
+        }
+
+
+        public int N => _n;
+        public double A => _a;
+        public double B => _b;
+
+        public System.Random RandomSource
+        {
+            get => _random;
+            set => _random = value ?? SystemRandomSource.Default;
+        }
+        /// <summary>
+        /// Gets the mean of the distribution.
+        /// </summary>
+        double IUnivariateDistribution.Mean => (_n * _a) / (_a + _b);
```

Comment:
What is the reasoning for choosing explicit interface implementations here, i.e. for making them unavailable on the type?